src/celeritas/em/distribution/UrbanMscScatter.hh

```diff
@@ -201,29 +204,30 @@ CELER_FUNCTION auto UrbanMscScatter::operator()(Engine& rng) -> MscInteraction
     real_type phi
         = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
 
+    MscInteraction result;
+    result.action = MscInteraction::Action::scattered;
+
     // Calculate displacement
-    Real3 displacement;
     if (input_.is_displaced && tau_ >= params_.tau_small)
     {
         // Sample displacement and adjust
-        displacement    = this->sample_displacement(rng, phi);
-        real_type rmax2 = (true_path_ - input_.geom_path)
-                          * (true_path_ + input_.geom_path);
-        real_type scaling
-            = this->calc_displacement_scaling(displacement, rmax2);
-        for (auto i : range(3))
+        real_type length = this->calc_displacement_length(
+            (true_path_ - input_.geom_path) * (true_path_ + input_.geom_path));
```

Comment:
Looks that this is based on the current master. As you suggested, input_ data member is removed in the fix-msc-rc branch and a new geom_path_ will be used instead of input_.geom_path.